crates/ast/src/ast_node.rs

```diff
@@ -80,6 +80,14 @@ pub fn children_nodes<TNode: AstNode, TChildNode: AstNode>(
     node.syntax().children().filter_map(TChildNode::cast)
 }
 
+/// 指定したノードから指定した`kind`のトークンを探して最初の一つ目を返します。
+pub fn token(parent: &SyntaxNode, kind: SyntaxKind) -> Option<SyntaxToken> {
+    parent
+        .children_with_tokens()
+        .filter_map(|it| it.into_token())
+        .find(|it| it.kind() == kind)
+}
```

Comment:
**Performance Issue**: The method 'token' retrieves a token of a specific kind from the syntax node's children, which may lead to performance issues if not properly optimized, especially if called frequently in a large AST structure. <br> **Fix**: Optimize the 'token' method by caching the results or restructuring how tokens are accessed to minimize traversal and lookup times. <br> **Code Suggestion**: 
 ```
 +    /// Optimized token retrieval with caching mechanism
 +    pub fn token(parent: &SyntaxNode, kind: SyntaxKind) -> Option<SyntaxToken> {
 +        static CACHE: HashMap<(SyntaxNode, SyntaxKind), Option<SyntaxToken>> = HashMap::new();
 +        if let Some(token) = CACHE.get(&(parent.clone(), kind)) {
 +            return token.clone();
 +        }
 +        let result = parent
 +            .children_with_tokens()
 +            .filter_map(|it| it.into_token())
 +            .find(|it| it.kind() == kind);
 +        CACHE.insert((parent.clone(), kind), result.clone());
 +        result
 +    }
 ```